datafusion/physical-expr/src/equivalence/class.rs

```diff
@@ -634,6 +634,18 @@ impl EquivalenceGroup {
             JoinType::RightSemi | JoinType::RightAnti => right_equivalences.clone(),
         }
     }
+
+    /// Checks if two expressions are equal either directly or through equivalence classes.
+    pub fn exprs_equal(
```

Comment:
This function does not work for two general expressions. In general, checking whether two expression trees are equal involves a lot of work: One not only needs to make sure all symbols (i.e. columns) are equivalent, but also take operator properties like commutativity into account.

For the purposes of this PR, I think it is sufficient to just make sure that:
1. `left` and `right` expression trees are "the same" except leaves.
2. Each leaf on the left and right side are equivalent w.r.t. some group.

Reply:
An easy way to test would be to look at `a + b` and `x + y` with `a` and `x` / `b` and `y` being equivalent.

Reply:
Resolved in 4d37673.